### PR TITLE
feat: add view code dialog

### PR DESCRIPTION
**Pull request by OEvgeny**
> Fixes #

## Changelog Entry

- Added code viewer dialog with syntax highlighting, in PR [#5335](https://github.com/microsoft/BotFramework-WebChat/pull/5335), by [@OEvgeny](https://github.com/OEvgeny)

## Description

Adds a dialog to view code snippets with syntax highlighting in chat messages. The feature focuses on improving code readability and maintaining good performance through selective language imports.

## Design

- Dialog opens when clicking the code button
- Uses Shiki for syntax highlighting with lazy loading
- Supports JavaScript, TypeScript, and Python initially
- Fetches code from the `basedOn`  if available in Message metadata

## Specific Changes

- Added `ActivityViewCodeButton` and `CodeContent` components
- Integrated Shiki with GitHub light/dark themes
- Added code icon button to message footer
- Added AI-generated code warning notice
- Added test coverage and localization strings

## -

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [] ~I have updated documentation~

## Review Checklist

> This section is for contributors to review your work.

-  [ ] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: __tests__/html/copyButton.hideAndShow.html [hunk 1]
```diff
@@ -53,6 +53,12 @@
 
         await pageConditions.numActivitiesShown(1);
 
+        await pageConditions.became(
+          'copy button is available',
+          () => document.querySelector(`[data-testid="${WebChat.testIds.copyButton}"]`) !== undefined,
```
Comment (compulim) [inline __tests__/html/copyButton.hideAndShow.html:58]: Should it be?

```
typeof document.querySelector() !== 'undefined'
```

Or

```
!!document.querySelector() // or just truthy/falsy is fine
```

File: packages/component/src/Attachment/Text/private/ActivityViewCodeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import classNames from 'classnames';
+import React, { memo, useCallback } from 'react';
+import useStyleSet from '../../../hooks/useStyleSet';
+import ActivityButton from './ActivityButton';
+import useShowModal from '../../../providers/ModalDialog/useShowModal';
+import CodeContent from './CodeContent';
+import LocalizedString from '../../../Utils/LocalizedString';
+import { hooks } from 'botframework-webchat-api';
```
Comment (compulim) [inline packages/component/src/Attachment/Text/private/ActivityViewCodeButton.tsx:8]: Sort.

File: packages/component/src/Attachment/Text/private/CodeContent.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}>;
+
+const highlighterPromise = createHighlighter();
+
+const CodeContent = memo(({ children, className, code, language, title }: Props) => {
+  const [highlightedCode, setHighlightedCode] = useState('');
+
+  useEffect(() => {
+    let mounted = true;
+    async function highlight() {
```
Comment (compulim) [inline packages/component/src/Attachment/Text/private/CodeContent.tsx:20]: Should this be IIFE?

File: packages/component/src/Attachment/Text/private/CodeContent.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      }
+      try {
+        const html = highlighter.codeToHtml(code, {
+          lang: language,
+          theme: 'github-light-default'
+        });
+
+        setHighlightedCode(html);
+      } catch (error) {
+        console.error('Failed to highlight code:', error);
```
Comment (compulim) [inline packages/component/src/Attachment/Text/private/CodeContent.tsx:33]:
```suggestion
        console.error('botframework-webchat: Failed to highlight code:', error);
```

File: packages/component/src/Attachment/Text/private/CodeContent.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+type Props = Readonly<{
+  children?: ReactNode | undefined;
+  className?: string | undefined;
+  code: string;
+  language?: string | undefined;
+  title: string;
+}>;
+
+const highlighterPromise = createHighlighter();
```
Comment (compulim) [inline packages/component/src/Attachment/Text/private/CodeContent.tsx:13]: Should this be lazy so it won't impact page load?

File: packages/component/src/Attachment/Text/private/CodeContent.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+          lang: language,
+          theme: 'github-light-default'
+        });
+
+        setHighlightedCode(html);
+      } catch (error) {
+        console.error('Failed to highlight code:', error);
+
+        const pre = document.createElement('pre');
+        pre.innerText = code;
```
Comment (compulim) [inline packages/component/src/Attachment/Text/private/CodeContent.tsx:36]: I guess `pre.textContent` is cheaper, but I am not exactly sure.

File: packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx [hunk 7]
```diff
@@ -22,6 +22,9 @@ import ActivityCopyButton from './ActivityCopyButton';
 import CitationModalContext from './CitationModalContent';
 import MessageSensitivityLabel, { type MessageSensitivityLabelProps } from './MessageSensitivityLabel';
 import isHTMLButtonElement from './isHTMLButtonElement';
+import ActivityViewCodeButton from './ActivityViewCodeButton';
```
Comment (compulim) [inline packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx:25]: Sort.

File: packages/component/src/Attachment/Text/private/CodeContent.tsx (hunk 5 above)
Comment (OEvgeny) [reply packages/component/src/Attachment/Text/private/CodeContent.tsx:13]: My concern is it then affects 1st modal opening. We already have everything loaded, maybe we could take some time to set it up as well. But I'm ready to change if you think it should be the other way around

Comment (compulim) [reply packages/component/src/Attachment/Text/private/CodeContent.tsx:13]: Maybe as a ShikiProvider later?

Comment (OEvgeny) [reply packages/component/src/Attachment/Text/private/CodeContent.tsx:13]: Yea, will make it a provider on the next iteration!

File: packages/component/src/Attachment/Text/private/ActivityViewCodeButton.tsx (hunk 2 above)
Comment (compulim) [reply packages/component/src/Attachment/Text/private/ActivityViewCodeButton.tsx]: Fixed.

**Review by compulim — DISMISSED** [review]

**Review by compulim — APPROVED** [review]

**Pull request merged**